src/jit/codegenarm64.cpp

```diff
@@ -2470,7 +2470,8 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
                 // We unconditionally throw a divide by zero exception
                 genJumpToThrowHlpBlk(EJ_jmp, SCK_DIV_BY_ZERO);
 
-                // We still need to call genProduceReg
+                // We still need to call genConsumeOperands/genProduceReg for liveness update.
+                genConsumeOperands(treeNode->AsOp());
```

Comment:
No, it is not OK to call genConsumeOperands() multiple times. It is doing a liveness update, and will assert if, for example, a last use is encountered for a variable that is not live. I was suggesting that you should encapsulate the part of genCodeForBinary() that does NOT include the genConsumeOperands() and then call it for your case, as well as from genCodeForBinary().